### DISCUSSION TITLE
Question on freezing layer

### DISCUSSION THREAD

**Discussion by Duwimpuhwe** [mrdbourke/pytorch-deep-learning#533] [Q&A]
I've replicated the codes for  ViT (per the paper replicating ) but freezing the layer why using the pretrained model is not working 

# Create ViT feature extractor model
import torchvision

# Download pretrained ViT model
vit_weights=torchvision.models.ViT_B_16_Weights.DEFAULT # "DEFAULT" means best available
pretrained_vit=torchvision.models.vit_b_16(weights=vit_weights)
# print(next(iter(pretrained_vit.parameters())))

# Freeze all layers in pretrained ViT model
for param in pretrained_vit.parameters():
  param.requres_grad=False # To be checked why it is not being changed to false
  

# Update the pretrained ViT head
embedding_dim=768 # ViT base
set_seeds()
pretrained_vit.heads=nn.Sequential(
    nn.LayerNorm(normalized_shape=embedding_dim),
    nn.Linear(in_features=embedding_dim,
              out_features=len(class_names)) 
)

# Get a summay
summary(model=pretrained_vit,
        input_size=(1,3,224,224),
        col_names=["input_size","output_size","num_params","trainable"],
        col_width=20,
        row_settings=["var_names"])

**Model summary**

![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/112572927/9d640934-93a3-48a4-b3f5-519abdd62092)
